src/components/CustomCollection/CustomCollection.test.tsx

```diff
@@ -221,10 +221,11 @@ describe('CustomCollection component', () => {
     scrollSpy.mockReset()
   })
 
-  it('renders the collection with DragDropContext', () => {
+  it('renders the collection with DragDropContext', async () => {
     const { container } = render(
       <CustomCollection {...exampleCollection} {...mockHandlers} />
     )
+    await screen.findByText('Example Collection')
```

Comment:
A lot of the changes in this file have to do with solving the `act` issues -- I do think it needs to be refactored in the future. A lot of functionality that is tested in this file may be better suited to other unit tests (and some duplicated at this point). But it's just too much to roll into this PR